### DISCUSSION TITLE
When to fork?

### DISCUSSION THREAD

**Discussion by Charlie-I-Am** [FIRST-Tech-Challenge/FtcRobotController#1002] [Q&A] [answered]
Hello, 

I noticed that updates happen over the summer and such, I was wondering when are updates guaranteed to be done for the next ftc season. And when would it be safe to start a new fork for the next season, because if we need to update the "main fork" we would have to take out our team code and then rewrite it. 

All in all, I am basically asking when is the cut off that updates will stop for the next season, is it the kickoff date, a different date, or is there none?

Thanks!

**Comment by ItsHarper** [comment, marked as answer]
Rather than just forking at a specific date, I would highly recommend using the approach documented here, which will allow you to update your Android Studio project without rewriting your code:

https://ftc-docs.firstinspires.org/en/latest/programming_resources/tutorial_specific/android_studio/fork_and_clone_github_repository/Fork-and-Clone-From-GitHub.html

That guide has specific instructions for how to update to the latest version of the SDK.

**Reply by Charlie-I-Am** [reply]
Thank you!
But just to make sure, the solution is that when there is an SDK update create a branch for it and then when ready merge it into the main, instead of rewriting the main?

**Reply by ItsHarper** [reply]
Yes, that's correct.
